### ISSUE TITLE
body parser doesn't treat content types as case-insensitive

### ISSUE THREAD

**Issue by robertklep** [restify/node-restify#877]
Example server:
```javascript
var restify = require('restify');
var server  = restify.createServer();

server.post('/', restify.jsonBodyParser(), function(req, res, next) {
  console.log('Body:', req.body);
  res.send(200);
  next();
});

server.listen(8080);
```

When called as follows, `req.body` unexpectedly is a `Buffer`:
```
$ curl -XPOST localhost:8080 -d '{"foo":"bar"}' -H'Content-Type:Application/Json'
```
(notice the content type case)

It works as expected with this:
```
$ curl -XPOST localhost:8080 -d '{"foo":"bar"}' -H'Content-Type:application/json'
```

According to [RFC2045](http://tools.ietf.org/html/rfc2045#section-5.1): _"Matching of media type and subtype is ALWAYS case-insensitive"_.

**Comment by yunong** [comment]
Thanks for finding this bug. Do you have the cycles to send a PR?

**Comment by yunong** [comment]
Fixed.

**Comment by yunong** [comment]
fixed as part of 395c042

**Comment by robertklep** [comment]
:+1: great work guys!
